### PR TITLE
Update `pytorch` version requirements in dev environments

### PR DESCRIPTION
**Pull request by dmitryduev**
Description
-----------
We've had an odd `pytorch` version requirement that has been making the CI happy, but me (and more recently also [the nightly suite](https://github.com/wandb/client/runs/6487772258)) -- unhappy.
Landed on the following solution:
- Followed [the official instructions](https://pytorch.org/get-started/locally/#start-locally) from `pytorch` :rofl:
- Used `cpu` for the compute platform everywhere by default (without pinning any particular version - we were pinning an outdated version in the tests) + `sed`'ing in `cu113` in the Dockerfile for the standalone GPU shard.

Also, while looking for a solution, stumbled upon the `wandb.sdk.internal.tpu.is_tpu_available` function, which unnecessarily imports and calls some `tf` functions -- refactored it, which yields a noticeable speed-up in `wandb.init`. 

Testing
-------
- Ran the full CI
- Fixed a Metaflow test
- Addressed a failure in `functional_tests/profiler/profiler.py`: `pytorch==1.11.0` (currently, latest) has a bug that sometimes causes a segfault on Linux, see `https://github.com/pytorch/pytorch/issues/69443`. They merged a fix, but it hasn't been released yet.
- Temporarily activated standalone testing for this PR, and it was green!
image

Checklist
-------
- Include reference to internal ticket "Fixes WB-NNNN" (and github issue "Fixes #NNNN" if applicable)

### CODE REVIEW

File: wandb/sdk/internal/tpu.py [hunk 1]
```diff
@@ -83,7 +83,12 @@ def start(self):
...
-def is_tpu_available():
+def is_tpu_available() -> bool:
+    tpu_name = os.environ.get("TPU_NAME", False)
+
+    if tpu_name is False:
+        return False
+
     try:
         from tensorflow.python.distribute.cluster_resolver import tpu_cluster_resolver  # type: ignore # noqa
         from tensorflow.python.profiler import profiler_client  # type: ignore # noqa
```
Comment (dmitryduev) [inline wandb/sdk/internal/tpu.py:94]: @vanpelt Chris, do you remember what these imports are for? Played safe and simply flipped the order of the checks, but still wondering if removing these for good is a good idea.

**Review by kptkin — APPROVED** [review]
Thanks for the fix!

File: .circleci/config.yml [hunk 2]
```diff
@@ -374,9 +374,9 @@ jobs:
       - run:
           name: Use py37
           command: |
-            pyenv install -s 3.7.0
+            pyenv install -s 3.7.9
```
Comment (kptkin) [inline .circleci/config.yml:377]: there are even later version of py37 if we want to update

File: functional_tests/lightning/train-gpu-ddp.yea [hunk 3]
```diff
@@ -3,7 +3,7 @@ plugin:
   - wandb
 tag:
   shard: service
-  skip: True
+  skip: true
```
Comment (kptkin) [inline functional_tests/lightning/train-gpu-ddp.yea:6]: soon... it will run in our gpu shard 🤩

File: functional_tests/profiler/profiler.py [hunk 4]
```diff
@@ -53,6 +53,7 @@ def train(data):
     with wandb.init() as run:
         run.config.id = "profiler_sync_trace_files"
         with torch.profiler.profile(
+            activities=[torch.profiler.ProfilerActivity.CPU],
```
Comment (kptkin) [inline functional_tests/profiler/profiler.py:56]: nice!

File: functional_tests/lightning/train-gpu-ddp.yea (hunk 3 above)
Comment (dmitryduev) [reply functional_tests/lightning/train-gpu-ddp.yea:6]: yea, I'm going to add a bunch of tests to our GPU shard tomorrow 🚗

File: .circleci/config.yml (hunk 2 above)
Comment (dmitryduev) [reply .circleci/config.yml:377]: well, I thought, something not too old and not too recent would be a good compromise.

**Pull request merged**